### PR TITLE
CU/8216-CleanUpTestIdPropsUsage

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Removed testIdProps usage and applied accessibility labels and testIDs where needed. I also went and combined some box props into text views to format things where I noticed them to reduce nesting.

| Screen or Component | What Changed | Tested on iOS | Tested on Android |
| ---------------------- | --------------- | -------------- | ------------------ |
| AccordionCollapsible  | updated to testId since a11yLabel wasn't used | [x] | [x] |
| BackButton                  | updated to a11yLabel | [x] | [x] |
| BaseListItem | updated to testId and a11yLabel | [x] | [x] |
| CarouselTabBar  | updated to testId since a11yLabel wasn't used  | [x] | [x] |
| CollapsibleView  | updated to a11yLabel since testID wasn't used  | [x] | [x] |
| BasicError  | updated to a11ylabel since testID wasn't used  | [x] | [x] |
| VASelector  | Updated to testID and a11yLabel for icon, testID for everything else  | [x] | [x] |
| List  | updated to testID and a11yLabel since both were used with same text | [x] | [x] |
| NavigationTabBar | updated to testID and a11yLabel since both were used with same text | [x] | [x] |
| Pagination | updated to testID and a11yLabel since both were used with same text  | [x] | [x] |
| VABulletList  | Updated to a11yLabel for TextView wrapper as testID wasn't used | [x] | [x] |
| VAImage | Updated to supply testID, was already supplying a11yLabel separetely | [x] | [x] |
| AppealIssues  | Removed as it wasn't used for either testID or a11yLabel  | [x] | [x] |
| TakePhotos  | updated to a11yLabel  | [x] | [x] |
| ClaimStatus | Removed as it wasn't used for either testID or a11yLabel, textID was supplied separately | [x] | [x] |
| ClaimPhase | change to a11yLabel  | [x] | [x] |
| EstimatedDecisionDate  | Changed to a11yLabel to combine the 2 text views to one a11yLabel  | [x] | [x] |
| ClaimsAndAppealsListView  | Removed as it wasn't used for either testID or a11yLabel  | [x] | [x] |
| DEPRECATED_ClaimsAndAppealsListView  | Removed as it wasn't used for either testID or a11yLabel  | [x] | [x] |
| NeedHelpData  | Removed all together from one textview, the call va was changed to a11yLabel | [x] | [x] |
| NoClaimsAndAppeals  |  Removed as it wasn't used for either testID or a11yLabel  | [x] | [x] |
| NoClaimsAndAppealsAccess  |  Removed as it wasn't used for either testID or a11yLabel  | [x] | [x] |
| NoDisabilityRatings  |  Removed as it wasn't used for either testID or a11yLabel  | [x] | [x] |
| GenericLetter  | Removed from template, updated textview to use a11yLabel  | [x] | [x] |
| LettersListScreen  | Removed as it wasn't used for either testID or a11yLabel | [x] | [x] |
| LettersOverviewScreen  | Removed as it wasn't used for either testID or a11yLabel, testID was already supplied separately  | [x] | [x] |
| NoLettersScreen  | Removed as it wasn't used for either testID or a11yLabel  | [x] | [x] |
| BiometricsPreferenceScreen | Removed from scrollview as it wasn't used and applied a11yLabel to text view since testID wasn't used  | [x] | [x] |
| NoAppointments  | Removed testID for both usages as it wasn't used, subText had an a11yLabel and applied that to the correct prop | [x] | [x] |
| NoMatchInRecords  | Removed as it wasn't used for either testID or a11yLabel | [x] | [x] |
| Folders  | Removed as it wasn't used for either testID or a11yLabel | [x] | [x] |
| NoInboxMessages  | Removed as it wasn't used for either testID or a11yLabel | [x] | [x] |
| NotEnrolledSM  | Removed as it wasn't used for either testID or a11yLabel | [x] | [x] |
| TermsAndConditions  | removed one unused testID/a11yLabel and replaced one with a11yLabel for the textA11y  | [x] | [x] |
| NoMilitaryInformationAccess | replaced with a11yLabel for the textA11y  | [x] | [x] |
| GenericOnboarding  | replaced with a11yLabel for the textA11y  | [x] | [x] |
| SplashScreen  | Replaced testIdProps with testID for e2e tests on scrollview that didn't use a11yLabel  | [x] | [x] |
| SyncScreen  | Removed all together, wasn't used for testId or A11yLabel  | [x] | [x] |
| WebviewControlButton  | Added a separate a11yLabel to the component props and supplied them both accordingly. Used by next component  | [x] | [x] |
| WebviewControls | Replaced testIdProps with both a11yLabel and testID as it was needed for both   | [x] | [x] |
| WebviewScreen  | Replaced testIdProps with testID for e2e tests/unit tests on the wrapper box and content. Neither used the a11yLabel   | [x] | [x] |
| WebviewTitle  | updated title to use a11yLabel for the box title wrapper with icon since testId wasn't used | [x] | [x] |
| LoginScreen  | Replaced testIdProps with testID for e2e tests on scrollview that didn't use a11yLabel, updated pressable with a11yLabel since testId wasn't used  | [x]  | [x]  |

## Screenshots/Video
N/A

## Testing
yarn test/detox

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
Screens should function and look the same, detox should still work, unit tests should still work and VQA should result in things staying the same.

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Review by alexandec — DISMISSED** [review]

**Comment by alexandec** [comment]
@Sparowhawk thanks for removing this finally! I'd recommend running the full set of E2E tests since many of them depend on TestIDs

**Comment by Sparowhawk** [comment]
@alexandec I ran the e2e tests with the help of @rbontrager to figure out which ones were actually broken and fixed them

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/NeedHelpData/NeedHelpData.tsx [hunk 1]
```diff
@@ -46,11 +46,7 @@ function NeedHelpData({ isAppeal }: NeedHelpDataProps) {
       <TextView variant="MobileBodyBold" accessibilityRole="header" accessible={true}>
         {t('claimDetails.needHelp')}
       </TextView>
-      <TextView
-        variant="MobileBody"
-        paragraphSpacing={true}
-        accessible={true}
-        accessibilityLabel={t('claimDetails.callVA.a11yLabel')}>
+      <TextView variant="MobileBody" mb={theme.dimensions.standardMarginBetween} accessible={true} accessibilityLabel={t('claimDetails.callVA.a11yLabel')}>
```
Comment (alexandec) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/NeedHelpData/NeedHelpData.tsx:49]: Linting error

**Review by alexandec — COMMENTED** [review]
Noticed a linting error and a conflict

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/NeedHelpData/NeedHelpData.tsx (hunk 1 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/NeedHelpData/NeedHelpData.tsx:49]: fixed

**Review by alexandec — DISMISSED** [review]

**Comment by Sparowhawk** [comment]
didn't see any comments on why this was moved back so assuming it was due to conflicts in the PR that I just fixed, moving back to with QA

**Review by alexandec — APPROVED** [review]

**Comment by rbontrager** [comment]
@Sparowhawk Couple things:
- The webview refresh button is no longer saying "refresh screen" button and now either says button refresh (ios) or just button (android).  
- Looking at the code (because I can't seem to find any examples in the app) it looks like the following might actually cause issues I'm not seeing.  Feel free to correct me if I'm wrong though.
     - BaseListItem a11ylabel has been changed from testID || '' to just testID
     - ListItem a11ylabel has been changed from titleA11yLabel ? titleA11yLabel : title to just titleA11yLabel

**Review by rbontrager — APPROVED** [review]

**Review by alexandec — APPROVED** [review]

**Pull request merged**